src/Router/DataObject/Route.php

```diff
@@ -0,0 +1,184 @@
+<?php
+declare(strict_types=1);
+
+namespace YapepBase\Router\DataObject;
+
+use YapepBase\Exception\IException;
+
+class Route
+{
+
+    /** @var string|null */
+    protected $name;
+
+    /** @var string */
+    protected $controller;
+
+    /** @var string */
+    protected $action;
+
+    /** @var string[] */
+    protected $methods;
+
+    /** @var string[]|null */
+    protected $regexPatterns;
+
+    /** @var array */
+    protected $unparsedPaths;
+
+    /** @var array */
+    protected $unparsedAnnotations;
+
+    /** @var Path[]|null */
+    protected $paths;
```

Comment:
Wy can't this be an empty array by default?

Reply:
So we can differentiate between something that has not been initialised and something that has, but has no paths... not that a route like that is useful, but not sure if it can be useful in certain cases, so didn't want to exclude that as a possibility.

Reply:
After the latest changes this can't be null any more, but forgot to initialise it and update the doc